messaging/kafka-producer/src/test/java/io/quarkus/ts/messaging/kafka/producer/SnappyCompressionIT.java

```diff
@@ -0,0 +1,162 @@
+
+    @KafkaContainer(vendor = KafkaVendor.STRIMZI)
+    static final KafkaService kafka = new KafkaService().withProperty("auto.create.topics.enable", "false");
+
+    @QuarkusApplication
+    static RestService app = new RestService().withProperty("kafka.bootstrap.servers", kafka::getBootstrapUrl);
+
+    @Test
+    public void checkCompressCodecSnappy() throws IOException, InterruptedException {
+        String msg = "This is the message";
+        GenericContainer<?> container = kafka.getPropertyFromContext(DockerContainerManagedResource.DOCKER_INNER_CONTAINER);
+        UniAssertSubscriber<Object> subscriber = makeHttpReqWithMessage("/messageEvent", msg)
+                .subscribe().withSubscriber(UniAssertSubscriber.create());
+
+        subscriber.awaitItem(Duration.ofSeconds(TIMEOUT_SEC)).getItem();
+        String logSegmentHead = container.execInContainer(COMMAND_LOG_CONTAINER)
+                .getStdout();
+        Assertions.assertTrue(logSegmentHead.contains("compresscodec: snappy"));
+    }
+
+    @Test
+    public void checkSnappyCompressionSize() throws IOException, InterruptedException {
+        GenericContainer<?> container = kafka.getPropertyFromContext(DockerContainerManagedResource.DOCKER_INNER_CONTAINER);
+        String largeMessage = "This is the large message".repeat(500);
+        int originalMessageSize = largeMessage.length();
+        UniAssertSubscriber<Object> subscriber = makeHttpReqWithMessage("/messageEvent", largeMessage)
+                .subscribe().withSubscriber(UniAssertSubscriber.create());
+
+        subscriber.awaitItem(Duration.ofSeconds(TIMEOUT_SEC)).getItem();
+        String logSegmentHead = container.execInContainer(COMMAND_LOG_CONTAINER)
+                .getStdout();
+        int compressedMessageSize = extractCompressedSize(logSegmentHead);
+        Assertions.assertTrue(compressedMessageSize < originalMessageSize,
+                "Compressed message size should be smaller than the original size");
+    }
+
+    @Test
+    public void checkIntegrityMessageAfterCompression() {
+        String msg = "This is the message";
+        KafkaConsumer<Integer, String> consumer = createConsumer();
+        UniAssertSubscriber<Object> subscriber = makeHttpReqWithMessage("/messageEvent", msg)
+                .subscribe().withSubscriber(UniAssertSubscriber.create());
+        subscriber.awaitItem(Duration.ofSeconds(TIMEOUT_SEC)).getItem();
+
+        ConsumerRecord<Integer, String> records = consumer.poll(Duration.ofMillis(10000))
+                .iterator().next();
+        Assertions.assertEquals(records.value(), msg);
+    }
+
+    @Test
+    public void sendBigJsonAndVerifyCompression() throws IOException, InterruptedException {
+        GenericContainer<?> container = kafka.getPropertyFromContext(DockerContainerManagedResource.DOCKER_INNER_CONTAINER);
+        int originalJsonSize = calculateOriginalJsonLength();
+        String largeJson = loadLargeJsonFromFile("/big_json.json");
```

Comment:
File big_json.json is referenced on multiple places. Put the file name in a constant.